crates/dt-cli/src/main.rs

```diff
@@ -77,12 +77,20 @@ fn main() {
         Command::Lint {} => todo!(),
     }
 
-    let (workspace_dir, toml_config) = TomlConfig::find_path(None)
-        .map_or((Path::new("."), None), |(workspace, toml)| {
-            (workspace, Some(TomlConfig::load(&toml).unwrap()))
-        });
+    let cwd = Path::new(".").canonicalize().unwrap();
+    let dir = Workspace::find_workspace_dir(&cwd);
+    let (config, workspace_dir) = (
```

Comment:
implement `Into<(Option<PathBuf>), &'start Path>` to avoid `dir` variable (we won't use result after workspace creation anyway)?

Reply:
I think currently it's alright.
Maybe you could try something like this if it feels cleaner:
```suggestion
    let (toml_config, workspace_dir) = match Workspace::find_workspace_dir(&cwd) {
      WorkspaceFindPathResult::TomlConfig { toml_file_path,
                workspace_dir, } => (Some(TomlConfig::load(toml_file_path)), workspace_dir)
      WorkspaceFindPathResult::LinuxMarker { workspace_dir }
      | WorkspaceFindPathResult::Fallback { workspace_dir } => (None, workspace_dir)
    };
    let config = ...
```